File: 프로그래머스/lv1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(int n, int m) {
    vector<int> answer;
    int x, y, r;
    x = n, y = m;
    while (y != 0) {
        r = x % y;
        x = y;
        y = r;
    }
    int gcd = x;
    int lcm = n * m / gcd;
    answer.push_back(gcd);
    answer.push_back(lcm);
    return answer;
}

File: 프로그래머스/lv1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.java
class Solution {
    public int[] solution(int n, int m) {
               
        int r, gcd, lcm;
        int temp1 = n,  temp2 = m;
                
        if(n < m){
            temp1 = m;
            temp2 = n;
        }
        
        while(temp2 != 0){
            r = temp1 % temp2;
            temp1 = temp2;
            temp2 = r;
        }
        gcd = temp1;
        lcm = n * m / gcd;
        
        int[] answer = {gcd, lcm};
        
        return answer;
    }
}